operator/pkg/object/objects_test.go

```diff
@@ -390,7 +390,7 @@ spec:
 	for _, tt := range parseYAMLToK8sObjectTests {
 		t.Run(tt.desc, func(t *testing.T) {
 			k8sObj, err := ParseYAMLToK8sObject([]byte(tt.objString))
-			if err != nil {
+			if err == nil {
```

Comment:
I think the test can be removed since it doesn't provide much value, and the parse logic is used in other tests as well.

Reply:
Just realized this function is a dup with `ParseK8sObjectsFromYAMLManifest` without checking the object validity. Maybe just remove this test, and make the function a wrapper of `ParseK8sObjectsFromYAMLManifest`.

Reply:
wrapped by ParseK8sObjectsFromYAMLManifest ？

Reply:
`ParseK8sObjectsFromYAMLManifest` parses several objects, and here `ParseYAMLToK8sObject` should be the same logic but with a condition "object number is equal to 1".